### ISSUE TITLE
[1.2.0] Graph Timespan is not used

### ISSUE THREAD

**Issue by jackouille-ch** [Cacti/cacti#1774]
Here is my configuration:
![image](https://user-images.githubusercontent.com/35721406/43473356-328b1a90-94f0-11e8-968a-64ac2b3113f0.png)

And while watching graphs:
![image](https://user-images.githubusercontent.com/35721406/43473437-69e4bed8-94f0-11e8-8c07-64d97e8e69bb.png)

I would like to have the 1 minute graph showing only the 4 last hours, currently it's nearly 10 hours.

Or it may be a mistake from my config...

**Comment by netniV** [comment]
Use the Graphs tab and set your timeframe's.

**Comment by jackouille-ch** [comment]
Yes of course I'm using this functionnality when I need it but here I was asking for setting it as default when I used the Time Graph View.
In fact I have increased my RRD in order to have more than 4 hours with the 1 minute polling interval but I would like to see only the 4 last hours (with 1 minute granularity), then 1 day at (with 5 minutes granularity).
And I just saw another bug:
![image](https://user-images.githubusercontent.com/35721406/43478535-6b444006-94fe-11e8-801b-b42dd078bece.png)

The Daily graph is displayed before the Hourly one, it should inversed, other graphs are displayed correctly

**Comment by cigamit** [comment]
You added rows to the RRA using the RRDtool tune command, that will leave those rows empty for a while.  Not much we can do about it without hacking the RRDtool dump output, which we are not to excited about.  Wish RRDtool would do that for us.  It'll be repaired by morning.

**Comment by jackouille-ch** [comment]
Yes of course I'm using this functionnality when I need it but here I was asking for setting it as default when I used the Time Graph View.
In fact I have increased my RRD in order to have more than 4 hours with the 1 minute polling interval but I would like to see only the 4 last hours (with 1 minute granularity), then 1 day at (with 5 minutes granularity).
And I just saw another bug:
![image](https://user-images.githubusercontent.com/35721406/43478535-6b444006-94fe-11e8-801b-b42dd078bece.png)

The Daily graph is displayed before the Hourly one, it should inversed, other graphs are displayed correctly

**Comment by jackouille-ch** [comment]
I think that there is an issue for graph generation:
![image](https://user-images.githubusercontent.com/35721406/43712589-fb55d5fc-9976-11e8-92a7-7734072ba371.png)

For me you should generate RRD graphs with Graph Timespan values:
Hourly : start=now-84h, end=now
Daily : start=now-1d, end=now
Weekly : start=now-1d, end=now
etc...

**Comment by jackouille-ch** [comment]
Still facing the same issue with Beta 1:
![image](https://user-images.githubusercontent.com/35721406/46600420-632f4380-caea-11e8-9d6b-09f8130aa71e.png)
So I should have a 4 hour graph with a 1 minute granularity but as written previously, defined Graph Timespan from DB is not used for generating graphs:
![image](https://user-images.githubusercontent.com/35721406/46600559-cf11ac00-caea-11e8-855c-75f7364dc85c.png)

**Comment by cigamit** [comment]
Oddly, it's rendering the entire time range.

**Comment by jackouille-ch** [comment]
Great thanks, I will upgrade to Beta 2 next week.
